test/experimental/features/retry_after/channel.go

```diff
@@ -96,14 +97,16 @@ func SendEvent(ctx context.Context, t *testing.T) *feature.Feature {
 	// Create The Base CloudEvent To Send (ID will be set by the EventsHub Sender)
 	event := cetest.FullEvent()
 
-	// Mark The Current Time Plus The Expected Retry-After Duration
-	earliestEventTime := time.Now().Add(time.Duration(retryAfterSeconds) * time.Second)
-
 	// Create A New Feature To Send An Event And Verify Retry-After Duration
 	f := feature.NewFeatureNamed("Send Events")
 	f.Setup("Install An EventsHub Sender", eventshub.Install(senderName, eventshub.StartSenderToResource(channel.GVR(), channelName), eventshub.InputEvent(event)))
 	f.Assert("Events Received", assert.OnStore(receiverName).MatchEvent(cetest.HasId(event.ID())).Exact(retryAttempts+1)) // One Successful Response
-	f.Assert("Event Timing Verified", assert.OnStore(receiverName).Match(receivedAfterEventInfoMatcher(earliestEventTime)).Exact(retryAttempts))
+
+	// expectedIntervalMargin is the acceptable duration between an expected retry being received and when
+	// it is being actually received. Manual tests indicate that this value is less than 20 ms, a bigger
+	// value is set to avoid flakyness.
+	expectedIntervalMargin := 500 * time.Millisecond
```

Comment:
(just a thought)
The only problem that I see here (anyway better than what we have today) is clock skew, I'm wondering if we should make it bigger or configurable.

For example, [in EC2, people have seen 5 minutes clock drift](https://stackoverflow.com/questions/14623650/how-to-deal-with-the-amazon-ec2s-clock-drift), so I'm wondering how reliable is 500ms :thinking:

Reply:
I think Clock drift under the second range should be a test pre-requisite if we want to test any time related feature.
Anyway, I'll take the margin up to `3s` and the retries parameter to `6s` if that's ok with you.

WDYT?

Reply:
At https://github.com/knative/eventing/pull/6128/commits/9a378b2e7b55a8a5679c6392f977bac4b4792527 I extracted the margin from the feature to the test to make it a bit easier to tweak if this is problematic.

Also the retries have been reconfigured to 6s with 3s drift tolerance.